reconcile/utils/terrascript_client.py

```diff
@@ -3691,3 +3695,54 @@ def populate_tf_resource_alb(self, resource, namespace_info,
 
         for tf_resource in tf_resources:
             self.add_resource(account, tf_resource)
+
+    def populate_tf_resource_secrets_manager(self, resource, namespace_info):
+        account, identifier, common_values, \
+            output_prefix, output_resource_name, annotations = \
+            self.init_values(resource, namespace_info)
+
+        tf_resources = []
+        self.init_common_outputs(tf_resources, namespace_info, output_prefix,
+                                 output_resource_name, annotations)
+
+        values = {
+            "name": identifier
+        }
+
+        region = common_values.get('region') or \
+            self.default_regions.get(account)
+        if self._multiregion_account_(account):
+            values['provider'] = 'aws.' + region
+
+        aws_secret_resource = aws_secretsmanager_secret(identifier, **values)
+        tf_resources.append(aws_secret_resource)
+
+        secret = common_values.get('secret')
+        secret_data = self.secret_reader.read_all(secret)
+
+        version_values = {
+            "secret_id": '${' + aws_secret_resource.id + '}',
+            "secret_string": json.dumps(secret_data, sort_keys=True)
+        }
+
+        if self._multiregion_account_(account):
+            version_values['provider'] = 'aws.' + region
+
+        aws_version_resource = \
+            aws_secretsmanager_secret_version(identifier, **version_values)
+        tf_resources.append(aws_version_resource)
+
+        # outputs
+        for k, v in secret_data.items():
+            output_name_0_13 = output_prefix + '__' + k
+            output_value = v
+            tf_resources.append(Output(output_name_0_13, value=output_value))
```

Comment:
why do you need to keep the secret information in terraform state?

Reply:
The intention is to send secret information to the namespace, in case the service can get it in cluster.

Reply:
if a service is running in a cluster, it is not likely to need to consume a secret from aws. this will be the case for openshift-vault-secrets. i'm not against this, but i do propose that we add this only when we find that it is needed. leaving it to you.